### PR TITLE
aaq urgency check endpoint V2

### PR DESCRIPTION
**Pull request by Hlamallama**
Search endpoint to call aaq seatch and urgency detect

### CODE REVIEW

File: aaq/views.py [hunk 1]
```diff
@@ -222,3 +223,29 @@ def search(request, *args, **kwargs):
...
+    serializer.is_valid(raise_exception=True)
+    message_text = serializer.validated_data["message_text"]
+    url = urllib.parse.urljoin(settings.AAQ_V2_API_URL, "/urgency-check-v2")
+    headers = {
+        "Authorization": settings.AAQ_V2_API_URL,
+        "Content-Type": "application/json",
+    }
+
+    response = requests.request("POST", url, json=message_text, headers=headers)
+    is_urgent = response.json()["is_urgent"]
```
Comment (erikh360) [inline aaq/views.py]: I think we should also return true/false to the rp flow, then there is no guessing what 0/1 means.

**Review by erikh360 — COMMENTED** [review]
Looking at the return body we could just return that straight to the flow.

File: aaq/views.py [hunk 2]
```diff
@@ -222,3 +223,21 @@ def search(request, *args, **kwargs):
...
+    serializer.is_valid(raise_exception=True)
+    message_text = serializer.validated_data["message_text"]
+    url = urllib.parse.urljoin(settings.AAQ_V2_API_URL, "/urgency-check-v2")
+    headers = {
+        "Authorization": settings.AAQ_V2_API_URL,
+        "Content-Type": "application/json",
+    }
+
+    response = requests.request("POST", url, json=message_text, headers=headers)
+    is_urgent = {"is_urgent": response.json()["is_urgent"]}
```
Comment (erikh360) [inline aaq/views.py]: We could just change this to:
`return Response(response.json(), status=status.HTTP_200_OK)`

File: aaq/tests/test_views.py [hunk 3]
```diff
@@ -527,3 +527,71 @@ def test_response_feedback_invalid_feedback_text_view(self):
...
         }
+
+
+class CheckUrgencyV2ViewTests(APITestCase):
+    url = reverse("aaq-check-urgency-v2")
+
+    @responses.activate
+    def test_urgency_check_urgent(self):
+        """
+        Test that we can get an urgency score of 1.0
```
Comment (erikh360) [inline aaq/tests/test_views.py]: Update these docstrings to match the true/false

File: aaq/tests/test_views.py [hunk 4]
```diff
@@ -527,3 +527,88 @@ def test_response_feedback_invalid_feedback_text_view(self):
...
+        """
+        user = get_user_model().objects.create_user("test")
+        self.client.force_authenticate(user)
+        fakeAaqUdV2Api = FakeAaqUdV2Api()
+        responses.add_callback(
```
Comment (erikh360) [inline aaq/tests/test_views.py]: Can you validate the request payload that is sent to the AAQ endpoint please?

Comment (Hlamallama) [reply aaq/tests/test_views.py]: i have this test test_urgency_check_invalid to check if the payload is valid

Comment (erikh360) [reply aaq/tests/test_views.py]: That is the payload to our endpoint, I'm talking about the payload e send to the aaq endpoint

File: aaq/tests/test_views.py [hunk 5]
```diff
@@ -527,3 +529,90 @@ def test_response_feedback_invalid_feedback_text_view(self):
...
+            responses.POST,
+            "http://aaq_v2/urgency-check-v2",
+            callback=fakeAaqUdV2Api.post_urgency_detect_return_true,
+            content_type="application/json",
+        )
+
+        payload = json.dumps({"message_text": "I am pregnant and out of breath"})
```
Comment (erikh360) [inline aaq/tests/test_views.py]: If you remove the .dumps here, you can remove the 2 .loads() calls on line 571

File: aaq/views.py [hunk 6]
```diff
@@ -173,21 +173,27 @@ def response_feedback(request, *args, **kwargs):
...
     }
+    search_url = urllib.parse.urljoin(settings.AAQ_V2_API_URL, "/search")
+
+    if "generate_llm_response" in serializer.validated_data:
+        search_payload["generate_llm_response"] = serializer.validated_data[
+            "generate_llm_response"
+        ]
+    if "query_metadata" in serializer.validated_data:
+        search_payload["query_metadata"] = serializer.validated_data["query_metadata"]
+
```
Comment (erikh360) [inline aaq/views.py]: Can we give these defaults in the serializer and always include them?

**Review by erikh360 — COMMENTED** [review]
Just the one comment for now.

Can we maybe move some of this functionality into a utils.py file to separate things a bit. Lets chat in slack about this

File: aaq/utils.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        "body": body_content,
+        "feedback_secret_key": feedback_secret_key,
+        "query_id": query_id,
+    }
+
+    urgency_check_payload = {
+        "message_text": query_text,
+    }
+
+    check_urgency_response = check_urgency_v2(urgency_check_payload)
```
Comment (erikh360) [inline aaq/utils.py]: Do you need to send a dict here? You build a dict inside `check_urgency_v2` too?

If the tests are passing it means somewhere we're not checking something

File: aaq/utils.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+    json_msg.update(check_urgency_response.data)
```
Comment (erikh360) [inline aaq/utils.py]: Can we use .json() instead of .data? Probably applies to a few places

**Review by erikh360 — COMMENTED** [review]
Starting to look good, just a few comments

**Review by erikh360 — APPROVED** [review]

**Pull request merged**